### DISCUSSION TITLE
Off-the-shelf State + IO + Either Monad

### DISCUSSION THREAD

**Discussion by glendonjklassen** [louthy/language-ext#1414] [General]
I apologize if I missed this somewhere in the repo/discussion, but is there something in the v5 pre-release that is an off-the-shelf IO/Either/State monad? I see that Eff is Readable but not Stateful. I'm relatively new to the library and FP so was trying to implement my own but the learning curve is hitting me pretty hard. I also see the above prototype but it looks like that's just in a not-for-consumption repo (even though it looks exactly like what I want).

Awesome project by the way. Thank you!

**Comment by aloslider** [comment]
I guess you want a [RWST](https://github.com/louthy/language-ext/discussions/1382). Also arcticles that might help: [1](https://paullouth.com/higher-kinds-in-csharp-with-language-ext-part-10-readert-monad-transformer/), [2](https://paullouth.com/higher-kinds-in-csharp-with-language-ext-part-11-statet-monad-transformer/) and [3](https://paullouth.com/higher-kinds-in-csharp-with-language-ext-part-12-writert-monad-transformer/)

**Reply by glendonjklassen** [reply]
I guess I can stack RWST with an Either and then that gets me what I need eh? I'm learning so some of the cognitive load of combining my own transformers seems high but I'll take a crack at it. Paul also had a smaller setup as a response on another thread that I may try. He requested I add this as a discussion thread of its own though.

**Reply by aloslider** [reply]
Ah, now I see he has already answered you with a detailed explanation.
> I'm learning so some of the cognitive load of combining my own transformers seems high

me too :)

It's quite hard to guess what problem you're trying to solve with this set of monads, so I thought RSWT fits your needs, but it would probably be easier to create your own one as Paul suggested. 
Is there something you struggle with exactly?
